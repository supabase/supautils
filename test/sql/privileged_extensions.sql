-- non-superuser extensions role
create role extensions_role login;
alter default privileges for role postgres in schema public grant all on tables to extensions_role;
set role extensions_role;
\echo

-- can create a privileged extension
create extension hstore;
select '1=>2'::hstore;

drop extension hstore;
\echo

-- custom scripts are run
select * from t2;

reset role;
drop table t2;
set role extensions_role;
\echo

-- custom scripts are run even for superusers
reset role;
create extension hstore;
drop extension hstore;
select * from t2;

drop table t2;
set role extensions_role;
\echo

-- cannot create other extensions
create extension moddatetime;
