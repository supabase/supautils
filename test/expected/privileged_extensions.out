-- non-superuser extensions role
create role extensions_role login;
alter default privileges for role postgres in schema public grant all on tables to extensions_role;
set role extensions_role;
\echo

-- can create a privileged extension
create extension hstore;
select '1=>2'::hstore;
  hstore  
----------
 "1"=>"2"
(1 row)

drop extension hstore;
\echo

-- custom scripts are run
select * from t2;
 column1 
---------
       1
(1 row)

reset role;
drop table t2;
set role extensions_role;
\echo

-- custom scripts are run even for superusers
reset role;
create extension hstore;
drop extension hstore;
select * from t2;
 column1 
---------
       1
(1 row)

drop table t2;
set role extensions_role;
\echo

-- cannot create other extensions
create extension moddatetime;
ERROR:  permission denied to create extension "moddatetime"
HINT:  Must be superuser to create this extension.
